src/App.tsx

```diff
@@ -15,36 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_LENGTH = 'Sort by length';
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, sortReverse }: { sortField: string; sortReverse: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField as string) {
+      case SORT_ALPHABETICALLY:
+        preparedGoods.sort((good1: string, good2: string) =>
+          good1.localeCompare(good2),
+        );
+        break;
+
+      case SORT_LENGTH:
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+
+      default:
+        return preparedGoods;
+    }
+  }
+
+  if (sortReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = React.useState('');
+  const [sortReverse, setSortReverse] = React.useState(false);
```

Comment:
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig
```suggestion
  const [isSortReversed, setIsSortReversed] = React.useState(false);
```